rtl/eclrtl/eclrtl.cpp

```diff
@@ -6066,7 +6066,13 @@ void rtlAddExceptionTag(StringBuffer & errorText, const char * tag, const char *
 void rtlSubstituteEmbeddedScript(size32_t &__lenResult, char * &__result, size32_t scriptChars, const char *script, size32_t outFieldsChars, const char *outFields, size32_t searchChars, const char *search)
 {
     StringBuffer result;
-    ::replaceString(result, rtlUtf8Size(scriptChars, script), script, rtlUtf8Size(searchChars, search), search, rtlUtf8Size(outFieldsChars, outFields), outFields);
+    bool foundMatch = false;
+    size_t sourceLen = rtlUtf8Size(scriptChars, script);
+    ::replaceString(result, sourceLen, script, rtlUtf8Size(searchChars, search), search, rtlUtf8Size(outFieldsChars, outFields), outFields, foundMatch);
```

Comment:
if this is the only place replaceString is called then I would consider a slightly different approach.
i) Pass a flag to indicate whether to copy if there is no match
ii) change the return type to be a boolean and return whether or not a replace occurred.

otherwise it is a slightly strange semantics for a pubic function.

Reply:
Changed to return a bool indicating if a match was found and takes a parameter that sets whether to copy regardless of a match being found.